### ISSUE TITLE
Eliza Agent Integration

### ISSUE THREAD

**Issue by dillchen** [hicommonwealth/commonwealth#10353]
### Description

Develop and deploy a system that integrates autonomous agents capable of interacting with users on platforms like Common and Farcaster. Enable these agents to perform onchain interactions, such as managing contests and tokens, and provide a user-friendly dashboard for agent management.

This is proposed / broken down into set of two tasks, and two milestones. The second milestone probably needs an additional person to work with. The

### **Acceptance Criteria**

1. **Milestone 1: Play with 5ish agents on Frick / Frack**
   - Development server is provisioned with SSH access.  
   - Common Client supports offchain message reading and writing.  
   - Approximately 5 agents are deployed and functional.

2. **Milestone 2: Allow (super)admins (and maybe small beta of folks to deploy agents) **
   - Super admin panel allows launching and editing up to 25 agents. (not worried about platform scalability
   - Onchain plugin is implemented for contests, coins, and namespaces.  
   - Agents can initiate contests, launch coins, and interact on Farcaster/Common as specified.

### Project Owner

@dillchen

#### **Tasks:**

1. [ ] **Development Environment Setup**  
   - Provision a development server (e.g., Vultr) with SSH access for the team and Eliza working on the Box.

2. [ ] **Build Common Client (eliza specific term) for Offchain Work**  
   - Develop a client to read and write messages, following existing patterns:  
     - Reference: [Eliza Client Farcaster](https://github.com/ai16z/eliza/tree/main/packages/client-farcaster)  
     - Documentation: [Eliza Clients](https://ai16z.github.io/eliza/docs/packages/clients/)

3. [ ] **Manual Agent Setup**  
   - Deploy approximately 5 agents and create / login with them onto Common:  
     - Posting content / react to mentions etc
     - Processing and responding to interactions  
     - Engaging users on platforms like **Common** and **Farcaster**.
     
4. [ ] **Super Admin Panel / Lite Dashboard: (could go into Milestone 2)**  
   - Develop a standalone dashboard (potentially on a separate domain) to:  
     - Launch new agents/characters.  
     - Edit character profiles without code modifications.  
     - Scope: Limit any scalability concerns to **25 agents** initially.
---
### **Milestone 2: Allow (super)admins (and maybe small beta of folks to deploy agents) **

Goal here is because [Patron AI, a potential internal project](https://www.notion.so/buildcommon/Patron-AI-Call-1531f6e2ed478042abdefe52969e2956?pvs=4) and [TinyAgents](https://github.com/SqueezeAILab/TinyAgent). Are already moving to a place where agents are deployed with nocode, via FC Mentions, are a few clicks on Common. We probably want to dev this but likely gonna be out of scope. We should discuss this after Milestone 1.

We probably would gate and or hardcode this to 25 - 100 agents as an "experiment" to get feedback from community

#### **Tasks:**

1. [ ] **Build Common Onchain Plugin**  
   - Extend the plugin framework to support Common onchain interactions, including:  
     - Contests  
     - Coin creation  
     - Namespace management  
   - References:  
     - [Eliza Plugins Documentation](https://ai16z.github.io/eliza/docs/packages/plugins/)  
     - [Coinbase Token Contract Plugin Example](https://ai16z.github.io/eliza/docs/packages/plugins/#7-coinbase-token-contract-plugin-elizaplugin-coinbase)

2. [ ] **Extend Agent Interactions**  
   - Enable agents to:  
     - Initiate contests.  
     - Launch coins directly from Farcaster/Common.  

3. [ ] **Scale it a bit horizontally?**  
   - Enable agents to:  
     - Initiate contests.  
     - Launch coins directly from Farcaster/Common.  

### **Additional Context**

- **Long-term Goals:**  
  - **AI Tagging:** Add an **AI Tag** to posts and user profiles in the UI.  
  - **Beta Launch for No-Code Agent Creation:**  
    - Build a beta feature allowing users to:  
      - Launch agents from the **Integrations Page**.  
      - Configure tools and plugins without coding.  
    - Enable users to manage multiple agents.  
  - **Expand Agent Functionality:**  
    - Agents should respond to posts seamlessly, similar to current behavior.

**References:**  
- [Eliza GitHub Repository](https://github.com/ai16z/eliza)  
- [Eliza Documentation](https://ai16z.github.io/eliza/)

**Support:** For assistance, refer to the [Discord channel](https://discord.com/invite/ai16z)

**Comment by dillchen** [comment]
Moved these out cause I don't like seeing them as tasks here

**Later Milestone 3: Allow (super)admins (and maybe small beta of folks to deploy agents) 

[Patron AI, a potential internal project](https://www.notion.so/buildcommon/Patron-AI-Call-1531f6e2ed478042abdefe52969e2956?pvs=4) and [TinyAgents](https://github.com/SqueezeAILab/TinyAgent) are already moving to a place where agents are deployed with nocode, via FC Mentions, or a few clicks on Common. We probably want to dev this but likely gonna be out of scope. We should discuss this after Milestone 1.

We probably would gate and or hardcode this to 25 - 100 agents as an "experiment" to get feedback from community. Because we're not committed to scaling, so whatever is MVP

**Potential Tasks:**

2. [ ] **Launch Agents via No code**  

3. [ ] **Scale it a bit horizontally?**  
   - Enable agents to:  
     - Initiate contests.  
     - Launch coins directly from Farcaster/Common.

**Comment by dillchen** [comment]
@ianrowan
